kong/utils.go

```diff
@@ -51,3 +52,36 @@ func stringArrayToString(arr []*string) string {
 	buf.WriteString(" ]")
 	return buf.String()
 }
+
+// headerRoundTripper injects Headers into requests
+// made via RT.
+type headerRoundTripper struct {
+	headers http.Header
+	rt      http.RoundTripper
+}
+
+// RoundTrip satisfies the RoundTripper interface.
+func (t headerRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	newRequest := new(http.Request)
+	*newRequest = *req
+	newRequest.Header = make(http.Header, len(req.Header))
+	for k, s := range req.Header {
+		newRequest.Header[k] = append([]string(nil), s...)
+	}
+	for k, v := range t.headers {
+		newRequest.Header[k] = v
+	}
+	return t.rt.RoundTrip(newRequest)
+}
+
+// RoundTripperWithHTTPHeaders returns a client which injects headers
+// before sending any request.
+func HTTPClientWitHeaders(client http.Client,
```

Comment:
I think this might be a typo, do we want `HTTPClientWithHeaders`? Otherwise -- this is useful!

Reply:
I also wonder if it might be useful to use the default http.Client if `client == nil` as well as a default transport.

Something like: 
```go
// HTTPClientWithHeaders returns a client which injects headers
// before sending any request.
func HTTPClientWithHeaders(client *http.Client,
	headers http.Header) http.Client {
	var res http.Client
	if client == nil {
		defaultTransport := http.DefaultTransport.(*http.Transport)
		res.Transport = defaultTransport
	} else {
		res = *client
	}
	res.Transport = headerRoundTripper{
		headers: headers,
		rt:      client.Transport,
	}
	return res
}
```

Otherwise, I think secure client initialization would look something like:

```go
	c := &http.Client{}
	defaultTransport := http.DefaultTransport.(*http.Transport)
	c.Transport = defaultTransport
	secureClient := HTTPClientWitHeaders(
		*c,
		http.Header{
			"kong-admin-token": []string{"my-token"},
		},
	)
	client, err := NewClient(nil, &secureClient)
```

Reply:
Excellent point. Thanks!
Updated the code and proceeding with merge now.